### PR TITLE
Generate functions for all elements with macro

### PR DESCRIPTION
**Pull request by peterstuart**

### CODE REVIEW

File: macros/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
```
Comment (peterstuart) [inline macros/Cargo.toml:1]: Procedural macros have to be in their own crate

File: macros/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+mod utils;
+
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use quote::quote;
+use syn::Expr;
+
+#[proc_macro]
+pub fn element(item: TokenStream) -> TokenStream {
+    let input = syn::parse_macro_input!(item as Expr);
```
Comment (peterstuart) [inline macros/src/lib.rs:19]: Fail if the input isn't an expression

File: macros/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    let value = match utils::str_literal(input) {
+        Ok(value) => value,
+        Err(error) => return error.into(),
+    };
```
Comment (peterstuart) [inline macros/src/lib.rs]: Fail if the input isn't a string literal

File: macros/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    let result = quote! {
+      pub fn #function_name(attributes: Vec<Attribute<Message>>, children: Vec<Node<Message>>) -> Node<Message> {
+        Node::Element(Element {
+          name: #node_name.into(),
+          attributes,
+          children,
+        })
+      }
+    };
```
Comment (peterstuart) [inline macros/src/lib.rs]: Generate a function with the given name.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -27,6 +27,12 @@ jobs:
       - name: test (lib)
         run: cargo test
 
+      - name: rustfmt (macros)
+        run: cd macros && cargo fmt -- --check
+        
+      - name: clippy (lib)
```
Comment (helixbass) [inline .github/workflows/main.yml]: Should be "clippy (macros)"?

**Review by helixbass — APPROVED** [review]
@peterstuart I didn't try and wrap my head around the macro implementation but looks nice

**Pull request merged**